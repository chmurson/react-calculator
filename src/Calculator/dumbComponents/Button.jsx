import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import './Button.styl'

export class Button extends React.Component {
  static propTypes = {
    onClick: PropTypes.func,
    buttonRef: PropTypes.func,
    className: PropTypes.string
  }

  static defaultProps = {
    className: '',
    onClick: () => null,
    buttonRef: () => null
  }

  onClick = e => {
    this.ref.blur()
    this.props.onClick(e)
  }

  setRef = ref => {
    if (ref) {
      this.ref = ref
    }
    this.props.buttonRef(ref)
  }

  ref

  render() {
    const { buttonRef, onClick, className, ...restOfProps } = this.props
    const finalClassName = classNames('custom-button', 'custom-button--regular', className)
    return <button tabIndex={-1} {...restOfProps} onClick={this.onClick} ref={this.setRef} className={finalClassName} />
  }
}
