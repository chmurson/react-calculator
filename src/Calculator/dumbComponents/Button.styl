.custom-button
  background-color white
  padding: 1
  border: 0
  &--active {
    background-color lightgray
  }
  &:focus
    padding 0
    border: 1px solid #0074D9
    outline none

.custom-button
  overflow: hidden

  padding: 24px

  cursor: pointer
  user-select: none
  transition: all 150ms linear
  text-align: center
  white-space: nowrap
  text-decoration: none !important

  color: #fff
  border: 0 none
  border-radius: 4px

  font-size: 13px
  font-weight: 500
  line-height: 1.3

  -webkit-appearance: none
  -moz-appearance: none
  appearance: none

  box-shadow: 2px 5px 10px var(--color-smoke)

  &:hover
    transition: all 150ms linear

    opacity: .85


  &:active, &--active
    transition: all 150ms linear
    opacity: .75


  &:focus
    outline: 1px dotted #959595
    outline-offset: -4px



.custom-button
  &--regular {
    color: #202129
    background-color: #f2f2f2

    &:hover {
      color: #202129
      background-color: #e1e2e2
      opacity: 1
    }

    &:active, &.custom-button--active {
      background-color: #d5d6d6
      opacity: 1
    }
  }
