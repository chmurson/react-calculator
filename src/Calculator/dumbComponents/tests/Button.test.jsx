import React from 'react'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import { Button } from '../Button'

describe('Button', () => {
  let component
  beforeAll(() => {
    component = mount(<Button>text</Button>)
  })

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
