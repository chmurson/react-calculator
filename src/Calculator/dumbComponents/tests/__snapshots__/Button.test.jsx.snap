// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<Button
  buttonRef={[Function]}
  className=""
  onClick={[Function]}
>
  <button
    className="custom-button custom-button--regular"
    onClick={[Function]}
    tabIndex={-1}
  >
    text
  </button>
</Button>
`;
